src/base_output_updater.cpp

```diff
@@ -36,6 +36,57 @@ void compute_delta_z_output(std::vector<float> &mu_a, std::vector<float> &var_a,
     }
 }
 
+void compute_delta_z_noise(std::vector<float> &mu_a, std::vector<float> &var_a,
```

Comment:
I would put comment for readability for each calculation section here. There are two cases: heteroscedastic and homoscedastic noises, which we want to estimate the noise distribution. This function only handle the heteroscedastic one. We need to add to the function in order to separate with the homosce ones